### PR TITLE
Changed Dockerfile base to dsci-100

### PR DESCRIPTION
**Pull request by ecool50**

### CODE REVIEW

**Comment by ttimbers** [comment]
What are the `--dev-build=False --minimize=False` arguments for?

**Comment by hfboyce** [comment]
> What are the `--dev-build=False` and `--minimize=False` arguments for?

Every time we try to build, it would timeout with `RuntimeError: JupyterLab failed to build`.

[This](`https://github.com/jupyterlab/jupyterlab/issues/7777 ) kinda explains a bit but over my understanding. After googling away without being able to build because it was timing out, several issues recommended these arguments to surpass this error. 

Also [here](https://pypi.org/project/jupyter-bokeh/) we found:
> On slow or limited memory systems, one can use --minimize=False to reduce compilation times or make Jupyter Lab build runtime bundles at all.

Essentially we are using them because if we don't, the build doesn't complete. 

We have checked that the container works, the assignments can be executed and the extensions used.

**Comment by ttimbers** [comment]
Thanks for the explanation. Let's try to remove it and see if it builds on the runner since it did last time with these extensions and you haven't changed anything with the extensions. If it fails, we will add them back in.

File: dockerfiles/dsci-011/Dockerfile [hunk 1]
```diff
@@ -1,42 +1,33 @@
...
-RUN pip install git+https://github.com/data-8/nbgitpuller \
-    && pip install git+https://github.com/altair-viz/altair_saver \
-    && jupyter serverextension enable --sys-prefix nbgitpuller \
-    && jupyter labextension install @jupyterlab/git \
-    && pip install jupyterlab-git \
-    && jupyter serverextension enable --py jupyterlab_git \
-    && pip install jupytext --upgrade \
-    && jupyter labextension install jupyterlab-flake8 \
-    && jupyter labextension install @ryantam626/jupyterlab_code_formatter \
+RUN jupyter labextension install --dev-build=False --minimize=False jupyterlab-flake8 \
```
Comment (ttimbers) [inline dockerfiles/dsci-011/Dockerfile:26]:
```suggestion
RUN jupyter labextension install jupyterlab-flake8 \
```

File: dockerfiles/dsci-011/Dockerfile [hunk 2]
```diff
@@ -1,42 +1,33 @@
...
+    && jupyter labextension install --dev-build=False --minimize=False @ryantam626/jupyterlab_code_formatter \
```
Comment (ttimbers) [inline dockerfiles/dsci-011/Dockerfile:27]:
```suggestion
    && jupyter labextension install @ryantam626/jupyterlab_code_formatter \
```

File: dockerfiles/dsci-011/Dockerfile [hunk 3]
```diff
@@ -1,42 +1,33 @@
...
     && pip install jupyterlab_code_formatter \
     && jupyter serverextension enable --py jupyterlab_code_formatter --sys-prefix\
-    && pip install black isort \
-    && jupyter lab build  \
-    && pip install flake8-nb
-RUN useradd -m -s /bin/bash -N -u 9999 jupyter
+    && jupyter lab build --minimize=False
```
Comment (ttimbers) [inline dockerfiles/dsci-011/Dockerfile:30]:
```suggestion
    && jupyter lab build
```

**Review by ttimbers — CHANGES_REQUESTED** [review]

**Comment by ttimbers** [comment]
I removed this flag and was able to successfully build the image locally after I upped the Docker memory allocation to 4GB from 2GB under Docker preferences:

Screen Shot 2020-11-19 at 11 15 29 AM

So I will now accept this PR and we can see if it builds successfully on the GitHubActions runner. Can you also try to build this version locally (without the  --dev-build=False and --minimize=False arguments) and test it again on the assignments? Just to be double sure?

**Comment by hfboyce** [comment]
Hi! Sorry, I missed this. I just rebuilt and tested it on the assignments and we are ok on my end here!

**Pull request merged**